vortex-mask/src/intersect_by_rank.rs

```diff
@@ -0,0 +1,86 @@
+use crate::Mask;
+
+impl Mask {
+    /// take the intersection of the `mask` with the set of true values in `self`.
+    ///
+    /// We are more interested in low selectivity `self` (as indices) with a boolean buffer mask,
+    /// so we don't optimize for other cases, yet.
+    pub fn intersect_by_rank(&self, mask: &Mask) -> Mask {
```

Comment:
This is a kinda hard to describe operation. I don't think the doc string really helped me. We kinda treat the left-hand-side as a lazily filtered mask?

Maybe an example is enough.

```suggestion
    /// so we don't optimize for other cases, yet.
    ///
    /// # Examples
    ///
    /// Keep the third and fifth set values from mask `m1`:
    /// ```
    /// let m1 = Mask::from_iter([true, false, false, true, true, true, false, true]);
    /// let m2 = Mask::from_iter([false, false, true, false, true]);
    /// assert_eq!(
    ///     m1.intersect_by_rank(m2),
    ///     Mask::from_iter([false, false, false, false, true, false, false, true])
    /// );
    /// ```
    pub fn intersect_by_rank(&self, mask: &Mask) -> Mask {
```

Reply:
if i'm reading this right, shouldn't the example result should be

```
[false, false, false, false, true, true, false, true]
```

?

Reply:
Maybe I misunderstand this function, but I thought the true count of the right-hand-side should match the true count of the output, which is two in the current example but three in your proposal.

Reply:
I misunderstood this function